.github/workflows/build.yml

```diff
@@ -0,0 +1,125 @@
+name: R Docker
+
+on:
+  push:
+  workflow_dispatch:
+    inputs:
+      variants:
+        description: |
+          Space-separated list of variants (platforms). Defaults to all variants.
+        required: false
+        default: ''
+        type: string
+      versions:
+        description: |
+          Space-separated list of R versions. Defaults to all versions.
+        required: false
+        default: ''
+        type: string
+      publish_images:
+        description: |
+          Publish images to Docker Hub?
+        required: false
+        default: false
+        type: boolean
+      include_patch_versions:
+        description: |
+          When building all R versions, include R patch version tags (x.y.z) as well?
+        required: false
+        default: false
+        type: boolean
+  workflow_call:
+    inputs:
+      versions:
+        description: |
+          Space-separated list of R versions. Defaults to all versions.
+        required: false
+        default: ''
+        type: string
+      publish_images:
+        description: |
+          Publish images to Docker Hub?
+        required: false
+        default: false
+        type: boolean
+      include_patch_versions:
+        description: |
+          Include R patch versions?
+        required: false
+        default: false
+        type: boolean
+    secrets:
+      DOCKERHUB_USERNAME:
+        required: true
+      DOCKERHUB_TOKEN:
+        required: true
+
+permissions:
+  contents: read
```

Comment:
Isn't this permission granted by default?

Reply:
It is, but this restricts the permissions from the default of read/write for all scopes: https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/

I've noticed more workflows adding this explicitly and think it's a good idea. It's also possible to restrict permissions on the repo level, but is less obvious and probably doesn't work with forks.

Reply:
It is, but this restricts the permissions from the default of read/write for all scopes: https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/

I've noticed more workflows adding this explicitly and think it's a good idea. It's also possible to restrict permissions on the repo level, but is less obvious and probably doesn't work with forks.